pennylane/_grad.py

```diff
@@ -21,55 +21,103 @@
 from autograd.numpy.numpy_boxes import ArrayBox
 from autograd.extend import vspace
 from autograd.wrap_util import unary_to_nary
+from pennylane.compiler import compiler
 
 make_vjp = unary_to_nary(_make_vjp)
 
 
 class grad:
-    """Returns the gradient as a callable function of (functions of) QNodes.
+    """A :func:`~.qjit` compatible gradient transformation that returns the gradient
+    as a callable function of (functions of) QNodes.
 
-    By default, gradients are computed for arguments which contain the property
-    ``requires_grad=True``. Alternatively, the ``argnum`` keyword argument can
-    be specified to compute gradients for function arguments without this property,
+    Inside a :func:`~.qjit` decorated program, this will patch to :func:`catalyst.grad` and
+    returns a callable object utilized by the supported compilers in
+    :doc:`Catalyst <catalyst:index>`.
```

Comment:
@maliasadi I would potentially remove this here, since we mention Catalyst already in a note below

Reply:
```suggestion
```

Reply:
Keeps it more concise!

Reply:
Sure, thanks @josh146!